finetune/adapter.py

```diff
@@ -195,7 +228,7 @@ def pad_right(x, pad_id):
     x = torch.stack([pad_right(x, pad_id=0) for x in input_ids])
     y = torch.stack([pad_right(x, pad_id=-1) for x in labels])
 
-    if isinstance(fabric.accelerator, MPSAccelerator):
+    if isinstance(fabric.accelerator, MPSAccelerator) or fabric.device.type == 'xla':
```

Comment:
We can simplify here:

```suggestion
    if fabric.device.type in ("mps", "xla"):
```